### PR TITLE
[mapcubed] Prototype "mapcubed" complementation (cubic data)

### PR DESCRIPTION
**Pull request by ehsteve**
This pull request creates a new kind of map object for stacks of images which are the same size (e.g. AIA cutouts, IRIS SJI images) over time. With this assumption, more functionality can be provided such as getting the mean of the maps or a light curve for a particular pixel.

This is a work in progress and the point of this pull request is to allow for a conversation of how to proceed and to allow for input from others.

### CODE REVIEW

File: sunpy/map/mapcubed.py [hunk 1]
```diff
@@ -0,0 +1,590 @@
...
+        """
+        return Map(np.min(self.data, axis=2), self._meta[self.ref_index])
+
+    def max(self):
+        """
+        Calculate the maximum value of the data array.
+        """
+        return Map(np.max(self.data, axis=2), self._meta[self.ref_index])
+
+    def plot(self, axes=None, resample=None, annotate=True, interval=200,
```
Comment (ehsteve) [inline sunpy/map/mapcubed.py]: there is a bug here. Need to copy latest code from original mapcube to make this work.

File: sunpy/map/mapcubed.py [hunk 2]
```diff
@@ -0,0 +1,590 @@
...
+
+    def submap(self, range_a, range_b, range_c=None):
+        """Returns a submap of the map with the specified range.
+        """
+        new_maps = []
+        for i in range(0, len(self)):
+            new_maps.append(self._get_map(i).submap(range_a, range_b))
+        return MapCubed(new_maps)
+
+    def lightcurve(self, location_a, location_b, range_c=None):
```
Comment (ehsteve) [inline sunpy/map/mapcubed.py]: The API here should be fixed. Would likely be better and more intuitive if a light curve were output with something like `mycubed[:,x,y]`

**Comment by ehsteve** [comment]
ping @wafels, @pymilo. Let's discuss this here.

**Comment by Cadair** [comment]
Does this currently enforce axes ordering? i.e. is it always [time, x, y] ?

**Pull request closed without merging**